contracts/libraries/Util.sol

```diff
@@ -71,12 +71,11 @@ library Util {
      * @param _data Passed data
      * @return bytes4 sig
      */
-    function getSig(bytes _data) internal pure returns (bytes4 sig) {
+    function getSig(bytes memory _data) internal pure returns(bytes4 sig) {
         uint len = _data.length < 4 ? _data.length : 4;
-        for (uint i = 0; i < len; i++) {
-            sig = bytes4(uint(sig) + uint(_data[i]) * (2 ** (8 * (len - 1 - i))));
+        for (uint256 i = 0; i < len; i++) {
+          sig |= bytes4(_data[i] & 0xFF) >> (i * 8);
```

Comment:
I'll take your word for it that this is equivalent ;-). I guess the tests would have broken if not.

Reply:
I got it right after a dozen tries lol. Nobody had manipulated strings like this in solidity 0.5 yet :).